docs/ORTModule_ModuleWithLoss_Wrapper.md

```diff
@@ -0,0 +1,114 @@
+# ONNX Runtime ModuleWithLoss Wrapper
+
+This document provides instructions on implementing a wrapper similar to the ModuleWithLoss Wrapper in Optimum. By implementing this wrapper, you can compute the loss inside ONNX Runtime (ORT), enabling you to leverage additional optimizations such as label sparsity optimization.
+
+#### ModuleWithLoss Wrapper
+
+- **Description**: The ModuleWithLoss wrapper is a class that extends the nn.Module class from PyTorch. It is designed to enhance the functionality of an existing model by allowing the computation of loss inside the wrapper, leveraging the benefits of ONNX Runtime (ORT) and enabling additional optimizations such as label sparsity optimization.
+
+These are the Major changes on the code, Refer to these snippets as a reference while writing your own implementation.
+
+```python
+class ModuleWithLoss(nn.Module):
+    def __init__(self, model, args, label_smoother):
+        super().__init__()
+        self._original_model = model
+        self.args = args
+        # Label smoothing
+        self.label_smoother = label_smoother
+
+    def forward(self, inputs: Dict[str, Union[torch.Tensor, Any]], return_outputs):
+        # The compute_model_plus_loss_internal is assigned once the class is instantiated.
+        # It should have same signature as Trainer.compute_loss().
+        # We do this to avoid potential un-synced states if we duplicated compute loss codes .
+        return self.compute_model_plus_loss_internal(self._original_model, inputs, return_outputs)
```

Comment:
this code is very specialized for optimum trainer, I would suggest not to just copy that as a common example. 

You can define a nn.Module which only compute stacks of layers, and return the outputs. the training script run the nn.Module's forward, and then feed its outputs into cross entropy loss, then trigger a backward. 

This is more like an example we want to cover using this doc.